packages/server/lib/middleware/access.middleware.ts

```diff
@@ -1,18 +1,49 @@
 import type { Request, Response, NextFunction } from 'express';
-import { isCloud, isBasicAuthEnabled, getLogger, metrics, stringifyError } from '@nangohq/utils';
+import type { Result } from '@nangohq/utils';
+import { isCloud, isBasicAuthEnabled, getLogger, metrics, stringifyError, Err, Ok } from '@nangohq/utils';
 import { LogActionEnum, ErrorSourceEnum, environmentService, errorManager, userService } from '@nangohq/shared';
 import db from '@nangohq/database';
 import * as connectSessionService from '../services/connectSession.service.js';
 import { NANGO_ADMIN_UUID } from '../controllers/account.controller.js';
 import tracer from 'dd-trace';
 import type { RequestLocals } from '../utils/express.js';
+import type { ConnectSession, DBEnvironment, DBTeam } from '@nangohq/types';
 
 const logger = getLogger('AccessMiddleware');
 
 const keyRegex = /^[0-9A-F]{8}-[0-9A-F]{4}-[4][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i;
 const ignoreEnvPaths = ['/api/v1/meta', '/api/v1/user', '/api/v1/user/name', '/api/v1/signin', '/api/v1/invite/:id'];
 
 export class AccessMiddleware {
```

Comment:
I wish all the middleware functions were composable but since we return early, sending 401 error it is not as easy as I would like to. I therefore extracted some some of the logic of the 2 middleware so it can be reused in a 3rd one.